src/vcpkg/base/system.process.cpp

```diff
@@ -708,26 +765,38 @@ namespace vcpkg
     {
         return cmd_execute(cmd_line, wd, get_clean_environment());
     }
+} // namespace vcpkg
 
+namespace
+{
 #if defined(_WIN32)
+    void close_handle_mark_invalid(HANDLE& target) noexcept
+    {
+        auto to_close = std::exchange(target, INVALID_HANDLE_VALUE);
+        if (to_close != INVALID_HANDLE_VALUE && to_close)
```

Comment:
Why is this checking `to_close` for null here -- and should that be a fatal error since it's an invariant violation?

Reply:
Some APIs use `0` as 'invalid handle', others use `INVALID_HANDLE_VALUE`. The intent is for this to work with either without having to worry about which is which.